internal/metrics/scraper.go

```diff
@@ -93,3 +102,33 @@ func (s *HTTPScraper) Scrape(ctx context.Context) (model.Vector, error) {
 
 	return allSamples, nil
 }
+
+type ObjectScraper struct {
+	gatherer prometheus.Gatherer
+}
+
+func NewObjectScraper(gather prometheus.Gatherer) Scraper {
+	return &ObjectScraper{gatherer: gather}
+}
+
+func (o *ObjectScraper) Scrape(_ context.Context) (model.Vector, error) {
+	metrics, err := o.gatherer.Gather()
+	if err != nil {
+		return nil, err
+	}
+	v := model.Vector{}
+	for _, m := range metrics {
+		metric := model.Metric{model.MetricNameLabel: model.LabelValue(*m.Name)}
+		for _, item := range m.Metric {
+			for _, labels := range item.Label {
+				metric[model.LabelName(*labels.Name)] = model.LabelValue(*labels.Value)
+			}
+			v = append(v, &model.Sample{Metric: metric, Value: model.SampleValue(*item.Counter.Value), Timestamp: model.Now()})
+		}
+	}
+	return v, nil
+}
+
+func (o *ObjectScraper) String() string {
```

Comment:
Not every type has or should have a string representation.  I think we should have called GetName or something like that. And what do u think about using a dynamic name? If I understand correctly, then we can have multiple object scrapers. Each with a different gatherer for a different feature. So perhaps there should be something unique for each instance

Reply:
In this case it's not the name that we're showing, but the targets: so for HTTP scrapers we'll return the URL, but for the only object scraper we have we just return a static string. I considered returning the name of scraper or something more meaningful, but since we only have 1 single object scraper I didn't see the point.
But if we reach the day where we have more object scrapers we'll definitely need to return a more representative value, rather than the constant.

Reply:
Note that the `targetName` will return `data transfer` in this case, so there is not much else we can add to this debug entry.

Reply:
I wonder. we use scrapers and not a single scraper. do we have targets with multiple URLs?

Reply:
We do target multiple URLs for a workload, but each scrape will target a single workload, so the wording `urls` here is misleading. I have reworded it a bit, as well as the value returned from the `String()` function for the HTTP scraper to improve the entry.

Reply:
Updated.